client/core/core.go

```diff
@@ -8146,19 +8155,26 @@ func deleteMatchFn(matchesFileStr string) (perMatchFn func(*db.MetaMatch, bool)
 	}, matchesFile.Close, nil
 }
 
+// archivedRecordsDataDirectory returns a data directory to save deleted archive
+// records.
+func (c *Core) archivedRecordsDataDirectory() string {
+	return filepath.Join(filepath.Dir(c.cfg.DBPath), "archived-records")
+}
+
 // DeleteArchivedRecords deletes archived matches from the database. Optionally
-// set a time to delete records after and file paths to save deleted records as
+// set a time to delete older records and file paths to save deleted records as
 // comma separated values. If a nil *time.Time is provided, current time is used.
-func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFile string) error {
+func (c *Core) DeleteArchivedRecords(olderThan *time.Time, saveMatchesToFile, saveOrdersToFile bool) error {
```

Comment:
re https://github.com/decred/dcrdex/pull/1916#discussion_r1005017938:

Recommend leaving this method alone. If you want different functionality, add a new method. I think for a first pass, we can just write to a system location like you've done, so I guess the new method is something like. 

```Go
func (c *Core) DeleteArchiveWithBackup(olderThan *time.Time, saveMatchesToFile, saveOrdersToFile bool) error {
	var matchesFile, ordersFile string
	if saveMatchesToFile {
		matchesFile = filepath.Join(c.archivedRecordsDataDirectory(), fmt.Sprintf("archived-matches-%d", time.Now().Unix()))
	}
	if saveOrdersToFile {
		ordersFile = filepath.Join(c.archivedRecordsDataDirectory(), fmt.Sprintf("archived-orders-%d", time.Now().Unix()))
	}
	return c.DeleteArchivedRecords(olderThan, matchesFile, ordersFile)
}
```

Reply:
That's an option.  In matrix I suggested also leaving the method alone, but instead detecting a `"."` as either file name as the indicator to use an automatic location in `datadir`.  A new method is OK too, but we can keep the interface smaller this way.

Reply:
> That's an option. In matrix I suggested also leaving the method alone, but instead detecting a `"."` as either file name as the indicator to use an automatic location in `datadir`. A new method is OK too, but we can keep the interface smaller this way.

That's fine too. Just don't wanna change the existing interface.

Reply:
I see we have a new `DeleteArchivedRecordsWithBackup` right now. Let's leave it unless other things come up in re-testing.

Reply:
> I see we have a new DeleteArchivedRecordsWithBackup right now.

Yes, I think 99% of users might not be comfortable with providing a non-existent file-path and that cuts out the process of dealing with `.` ,`./`. The `cmdSummary` for existing RPC `deleteArchivedRecordsRoute` says `that file locations are from the perspective of dexc and not the caller` so it makes sense to use the `dexc` `datadir` without having to require it from the user who might not be on the machine running `dexc` atm.